20230727/s10-1690466917636/README.md

```diff
@@ -0,0 +1,87 @@
+LOG_LEVEL=error
+## Added for Test Scenario 5
+EVENT_SDK_ASYNC_OVERRIDE_EVENTS=""
+EVENT_SDK_LOG_FILTER=""
+## Added for Test Scenario 7
+UV_THREADPOOL_SIZE=16
+ACCOUNT_LOOKUP_SERVICE_VERSION=v14.2.3
+## Changes for Test Scenario 9
+Enabled in-memory storage for Mysql ALS with following config in docker-compose file
+## Changes for Test Scenario 10
+Disabled JSON.stringify in logResponse function of ALS
+```
+
+## Environment
+
+- m6i.2xlarge
+  - 8 CPU - 3.5 GHz 3rd Generation Intel Xeon Scalable processors (Ice Lake 8375C)
+  - 32gb RAM
+- https://gist.github.com/mdebarros/6d9ac90f33c96031cbce6b9a3ea8048e
+
+## k6 Test Config
+
+```json
+{
+  "scenarios": {
+    "accountLookup": {
+      "executor": "ramping-vus",
+      "exec": "accountLookupScenarios",
+      "startVUs": 1,
+      "stages": [
+        { "duration": "2m", "target": 1 },
+        { "duration": "5m", "target": 1 }
+      ]
+    }
+  },
+  "thresholds": {
+    "iteration_duration": [ "p(95)<1000" ],
+    "http_req_failed": [ "rate<0.01" ],
+    "http_req_duration": [ "p(95)<1000" ]
+  }
+}
+```
+
+## Snapshots
+
+N/A
+
+## Observations
+
+- `Account-Lookup-Service` and the `Callback-Handler` Simulator Service are able to handle `100 Ops/s` End-to-end, while sustaining an average duration of around `14ms`. This is shown by the following dashboards/metrics:
+  - [K6](./images/Official%20k6%20Test%20Result.png)
+    - `Iteration Rate` (Mean) = `100 Ops/s`
+    - `Ieration Duration (avg)` (Mean) = `14ms`
+  - [Callback Handler Svc](./images/Supporting%20Services%20-%20Callback%20Hander%20Service.png)
+    - `op:fspiop_put_parties_end2end - success:true` - observe the `E2E, Request, Response Calculations Processed Per Second` Graph. Note the Mean includes the pre/post run.
+    - `op:fspiop_put_parties_end2end - success:true` - observe the `E2E, Request, Response Performance Timing Calculations` fall-inline with the observed duration.
+    - The `op:fspiop_put_parties_request` and `op:fspiop_put_parties_response` fall-inline with the observations.
+  - [Account-Lookup-Service](./images/dashboard-account-lookup-service.png)
+    - Egress and Ingress metrics for `getPartiesByTypeAndId` and `putPartiesByTypeAndId` are similar for both `duration` and `Op/s`.
+    - The `validateParticipant`'s `duration` is @ `1.32ms`, vs `30ms`...which is in-line with other egress.
+    - The `validateParticipant`'s `Op/s` is @ near `300 Op/s`, vs `30`...which is in-line with `Scenario #2` and the three-fold increase over the End-to-end `100 Op/s`.
+  - [Docker Node Monitoring](./images/docker-prometheus-monitoring.png)
+    - `Account-Lookup-Service` is showing a reduced CPU usage of `70%`, `30%` down from `100%`, indicating that the service is able to do more if K6 VUs are increased.
+    - `Callback-Handler` is within bounds of the `Scenario #1`.
+
+- `JSON.stringify` is a demanding operation which is causing a bottleneck on the NodeJS `Event-loop` impacting the End-to-end `Op/s` and `duration`.
```

Comment:
It is also impacting HTTP Sync response which is a major issue than end-to-end duration. This eventually causing K8S pod restarts because of the http health endpoint unavailability because of event loop.

Reply:
Done